subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc

```diff
@@ -1048,3 +1048,40 @@ Alternatively, for Kotlin implementations, the Kotlin compiler can be configured
 include::sample[dir="snippets/configurationCache/samConversion/groovy",files="build.gradle[tags=configure]"]
 include::sample[dir="snippets/configurationCache/samConversion/kotlin",files="build.gradle.kts[tags=configure]"]
 ====
+
+[[config_cache:not_yet_implemented:accessing_top_level_at_execution]]
+=== Accessing top-level methods and variables of a build script at execution time
+
+Currently, Gradle cannot run a task restored from the configuration cache if a task action calls a method defined directly in the build script.
+
+In case of the Groovy DSL, the action fails when running from the cache with an error message about method not being found.
+In the following snippet, the task `listFiles` uses such method:
+====
+include::sample[dir="snippets/configurationCache/topLevel/groovy",files="build.gradle[tags=not-supported]"]
+====
+
+Running the task the second time when the configuration cache is reused gives an error:
+----
+include::{snippetsPath}/configurationCache/topLevel/tests-groovy/notSupported.out[]
+----
+
+The workaround is to convert the method to the static method of a proper class:
+====
+include::sample[dir="snippets/configurationCache/topLevel/groovy",files="build.gradle[tags=workaround]"]
+====
+
+In case of the Kotlin DSL, the task cannot be stored to the configuration cache at all, because the captured script object reference cannot be serialized.
+For the same reason, it is not possible to reference a top-level variable from the task action too.
```

Comment:
```suggestion
Build scripts written in Kotlin cannot store tasks that reference top-level methods or variables in the configuration cache at all.
This limitation exists because the captured script object references cannot be serialized.
As a result, the Kotlin version of this task won't fail when re-run using the configuration cache.
But it won't be stored in the configuration cache, which slows down the build.
```
Suggestion: significant rewrite, eliminating the last sentence entirely. Rephrase the introduction for (near) parallelism to the Groovy reference above. Split the effect/cause apart to make the limitation clearer, and allow readers to "tune out" from the explanation if they don't understand the (quite technical) reasoning behind it.
Make the conclusion clearer -- at first, I thought you meant that *no* runs with this task would succeed with the configuration cache enabled. I *think* the suggested sentences are correct, but you should double check.